sci-libs/mkl/mkl-10.0.5.025-r1.ebuild

```diff
@@ -0,0 +1,371 @@
+			includedir=\${prefix}/include
+			Name: ${x}
+			Description: Intel(R) Math Kernel Library implementation of ${x}
+			Version: ${PV}
+			URL: ${HOMEPAGE}
+		EOF
+	done
+	cat >> blas-${prof}.pc <<-EOF
+		Libs: -Wl,--no-as-needed -L\${libdir} ${2} ${3} -lmkl_core ${4} -lpthread
+	EOF
+	cat >> cblas-${prof}.pc <<-EOF
+		Requires: blas
+		Libs: -Wl,--no-as-needed -L\${libdir} ${2} ${3} -lmkl_core ${4} -lpthread
+		Cflags: -I\${includedir}
+	EOF
+	cat >> lapack-${prof}.pc <<-EOF
+		Requires: blas
+		Libs: -Wl,--no-as-needed -L\${libdir} ${2} ${3} -lmkl_core -lmkl_lapack ${4} -lpthread
+	EOF
+	insinto ${MKL_LIBDIR}
+	for x in blas cblas lapack; do
+		doins ${x}-${prof}.pc
+		cp eselect.${x} eselect.${x}.${prof} || die
+		echo "${MKL_LIBDIR}/${x}-${prof}.pc /usr/@LIBDIR@/pkgconfig/${x}.pc" \
+			>> eselect.${x}.${prof}
+		eselect ${x} add $(get_libdir) eselect.${x}.${prof} ${prof}
+	done
+}
+
+mkl_make_profiles() {
+	local clib="gf"
+	has_version 'dev-lang/ifc' && clib+=" intel"
+	local slib="-lmkl_sequential"
+	local rlib="-liomp5"
+	local pbase=${PN}
+	local c
+	for c in ${clib}; do
+		local ilib="-lmkl_${c}_lp64"
+		use x86 && ilib="-lmkl_${c}"
+		local tlib="-lmkl_${c/gf/gnu}_thread"
+		local comp="${c/gf/gfortran}"
+		comp="${comp/intel/ifort}"
+		mkl_add_profile ${pbase}-${comp} ${ilib} ${slib}
+		mkl_add_profile ${pbase}-${comp}-threads ${ilib} ${tlib} ${rlib}
+		if use int64; then
+			ilib="-lmkl_${c}_ilp64"
+			mkl_add_profile ${pbase}-${comp}-int64 ${ilib} ${slib}
+			mkl_add_profile ${pbase}-${comp}-threads-int64 ${ilib} ${tlib} ${rlib}
+		fi
+	done
+}
+
+src_install() {
+	dodir ${MKL_DIR}
+
+	# install license
+	if  [[ ! -f ${INTEL_LIC_DIR}/${MKL_LIC} ]]; then
+		insinto ${INTEL_LIC_DIR}
+		doins "${WORKDIR}"/${MKL_LIC} || die "install license failed"
+	fi
+
+	# install main stuff: cp faster than doins
+	einfo "Installing files..."
+	local cpdirs="benchmarks doc examples include interfaces lib man tests"
+	local doinsdirs="tools"
+	cp -pPR ${cpdirs} "${D}"${MKL_DIR} \
+		|| die "installing mkl failed"
+	insinto ${MKL_DIR}
+	doins -r ${doinsdirs} || die "doins ${doinsdirs} failed"
+	dosym mkl_cblas.h ${MKL_DIR}/include/cblas.h
+
+	# install blas/lapack profiles
+	mkl_make_generic_profile
+	mkl_make_profiles
+
+	# install env variables
+	cat > 35mkl <<-EOF
+		MKLROOT=${MKL_DIR}
+		LDPATH=${MKL_LIBDIR}
+		MANPATH=${MKL_DIR}/man
+	EOF
+	doenvd 35mkl
+}
+
+pkg_postinst() {
+	# if blas profile is mkl, set lapack and cblas profiles as mkl
+	local blas_prof=$(eselect blas show | cut -d' ' -f2)
+	local def_prof="mkl-gfortran-threads"
+	has_version 'dev-lang/ifc' && def_prof="mkl-ifort-threads"
+	use int64 && def_prof="${def_prof}-int64"
+	for x in blas cblas lapack; do
+		local cur_prof=$(eselect ${x} show | cut -d' ' -f2)
+		if [[ -z ${cur_prof} ||	${cur_prof} == ${def_prof} ]]; then
+			# work around eselect bug #189942
+			local configfile="${ROOT}"/etc/env.d/${x}/$(get_libdir)/config
+			[[ -e ${configfile} ]] && rm -f ${configfile}
```

Comment:
missing `|| die` (this will require restructuring, you cannot use a short-form bash if for this)